sqlobject/src/main/java/org/jdbi/v3/sqlobject/BatchHandler.java

```diff
@@ -36,65 +32,61 @@
-        this.sql = SqlAnnotations.getSql(anno, raw_method);
+        SqlBatch anno = method.getAnnotation(SqlBatch.class);
+        this.sql = SqlAnnotations.getSql(anno, method);
         this.transactional = anno.transactional();
-        this.batchChunkSize = determineBatchChunkSize(sqlObjectType, raw_method);
+        this.batchChunkSize = determineBatchChunkSize(sqlObjectType, method);
     }
 
-    private ChunkSizeFunction determineBatchChunkSize(Class<?> sqlObjectType, Method raw_method)
+    private ChunkSizeFunction determineBatchChunkSize(Class<?> sqlObjectType, Method method)
     {
         // this next big if chain determines the batch chunk size. It looks from most specific
         // scope to least, that is: as an argument, then on the method, then on the class,
         // then default to Integer.MAX_VALUE
 
-        int index_of_batch_chunk_size_annotation_on_parameter;
-        if ((index_of_batch_chunk_size_annotation_on_parameter = findBatchChunkSizeFromParam(raw_method)) >= 0) {
-            return new ParamBasedChunkSizeFunction(index_of_batch_chunk_size_annotation_on_parameter);
+        int batchChunkSizeParameterIndex;
+        if ((batchChunkSizeParameterIndex = indexOfBatchChunkSizeParameter(method)) >= 0) {
+            return new ParamBasedChunkSizeFunction(batchChunkSizeParameterIndex);
         }
-        else if (raw_method.isAnnotationPresent(BatchChunkSize.class)) {
-            final int size = raw_method.getAnnotation(BatchChunkSize.class).value();
+        else if (method.isAnnotationPresent(BatchChunkSize.class)) {
+            final int size = method.getAnnotation(BatchChunkSize.class).value();
             if (size <= 0) {
                 throw new IllegalArgumentException("Batch chunk size must be >= 0");
             }
             return new ConstantChunkSizeFunction(size);
         }
         else if (sqlObjectType.isAnnotationPresent(BatchChunkSize.class)) {
-            final int size = BatchChunkSize.class.cast(sqlObjectType.getAnnotation(BatchChunkSize.class)).value();
+            final int size = sqlObjectType.getAnnotation(BatchChunkSize.class).value();
             return new ConstantChunkSizeFunction(size);
         }
         else {
             return new ConstantChunkSizeFunction(Integer.MAX_VALUE);
         }
     }
 
-    private int findBatchChunkSizeFromParam(Method raw_method)
+    private int indexOfBatchChunkSizeParameter(Method method)
     {
-        Annotation[][] param_annos = raw_method.getParameterAnnotations();
-        for (int i = 0; i < param_annos.length; i++) {
-            Annotation[] annos = param_annos[i];
-            for (Annotation anno : annos) {
-                if (anno.annotationType().isAssignableFrom(BatchChunkSize.class)) {
-                    return i;
-                }
+        Annotation[][] parameterAnnotations = method.getParameterAnnotations();
+        for (int i = 0; i < parameterAnnotations.length; i++) {
```

Comment:
If you're going to do streams, might as well go whole hog?  I'm not sure the single element per Stream approach is idiomatic...

```java
return IntStream.range(0, parameterAnnotations.length)
                          .filter(i -> parameterAnnotations[i].annotationType().isAssignableFrom(BatchChunkSize.class))
                          .findFirst().orElse(-1);
```